### ISSUE TITLE
test_filter not being correctly applied when there are multiple test cases (go)

### ISSUE THREAD

**Issue by kriscfoster** [bazelbuild/intellij#3691]
Repository with reproduction: https://github.com/kriscfoster/bazel-intelij-go-test-filter-repro

## command line

(only the desired test case runs)

```
➜ bazel test //... --test_filter=TestMultipleCases/TestCase1 
...
==================== Test output for //:example_test:
=== RUN   TestMultipleCases
2022/06/16 07:15:01 running test: TestCase1
=== RUN   TestMultipleCases/TestCase1
--- PASS: TestMultipleCases (0.00s)
    --- PASS: TestMultipleCases/TestCase1 (0.00s)
PASS
...
```

## using intelij plugin

(undesired test cases also run)

<img width="842" alt="Screenshot 2022-06-16 at 08 21 01" src="https://user-images.githubusercontent.com/17026751/174014898-0832bfed-fa38-4c21-82fe-e7ef2887118c.png">

The correct `--test_filter` isn't applied & more than the desired tests run:

```
/Users/kfoster/bin/bazel test --tool_tag=ijwb:IDEA:ultimate ... --test_filter=^TestMultipleCases$ -- //:example_test
...
==================== Test output for //:example_test:
=== RUN   TestMultipleCases
2022/06/16 07:18:22 running test: TestCase1
2022/06/16 07:18:22 running test: TestCase2
=== RUN   TestMultipleCases/TestCase1
=== RUN   TestMultipleCases/TestCase2
--- PASS: TestMultipleCases (0.00s)
    --- PASS: TestMultipleCases/TestCase1 (0.00s)
    --- PASS: TestMultipleCases/TestCase2 (0.00s)
PASS
================================================================================
Target //:example_test up-to-date:
```

**Comment by JamesRTaylor** [comment]
Would love to see a fix for this. Happy to be a guinea pig for testing.

**Comment by blorente** [comment]
Hello! I'm working on this now. I've opened a PR with _some_ fixes for this, but the full solution requires semantic information, which I'm not sure how to get. Will keep plugging at it :)

**Comment by blorente** [comment]
This should be solved by #4131. I've changed it so that I expect all cases work now.
